logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/conversation/SyncConversationsUseCaseTest.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.data.conversation.Conversation
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.SystemMessageBuilder
+import com.wire.kalium.logic.framework.TestConversation
+import com.wire.kalium.logic.functional.Either
+import io.mockative.any
+import io.mockative.eq
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.thenDoNothing
+import io.mockative.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class SyncConversationsUseCaseTest {
+
+    @Test
+    fun testGiven_whenInvoked_thenFetchConversations() = runTest {
+
+        val (arrangement, useCase) = Arrangement()
+            .withGetConversationsIdsReturning(emptyList())
+            .withFetchConversationsSuccessful()
+            .arrange()
+
+        useCase.invoke()
+
+        verify(arrangement.conversationRepository)
+            .suspendFunction(arrangement.conversationRepository::fetchConversations)
+            .wasInvoked(exactly = once)
+    }
+
+    @Test
+    fun testGivenProtocolChanges_whenInvoked_thenInsertHistoryLostSystemMessage() = runTest {
```

Comment:
```suggestion
    fun givenProtocolChanges_whenInvoked_thenInsertHistoryLostSystemMessage() = runTest {
```

Reply:
iOS habits where tests methods needs to be prefixed with `test`

Reply:
Updated